### ISSUE TITLE
BUG: Qt5Agg window size regression

### ISSUE THREAD

**Issue by larsoner** [matplotlib/matplotlib#15611]
### Bug report

**Bug summary**

`figsize` no longer works properly on Qt5Agg backend. By `git bisect` it worked on c19ebd6d6705cd845b6c69a1a19595b25825bf2e but not e6367c9c40419249e55d80a2806d17fd11137f2b. It looks like this might have been a direct push and not a PR (?).

Doing `git revert e6367c9c40419249e55d80a2806d17fd11137f2b` on `master` fixes the problem.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
fig = plt.figure(figsize=(5, 20))
```

**Actual outcome**

Window is not the full vertical height of my screen:

<details>

![Screenshot from 2019-11-06 10-32-24](https://user-images.githubusercontent.com/2365790/68312374-f2f24900-0080-11ea-9970-d44169d5864e.png)

</details>

**Expected outcome**

Taller window:

<details>

![Screenshot from 2019-11-06 10-51-40](https://user-images.githubusercontent.com/2365790/68313973-71e88100-0083-11ea-9c52-02689e098a5b.png)

</details>

**Matplotlib version**

  * Operating system: Ubuntu x86_64
  * Matplotlib version: `master` back to e6367c9c40419249e55d80a2806d17fd11137f2b
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.8.0
  * Other libraries: PyQt5 5.13.2

Everything installed from `pip` (other than `matplotlib`, which I build locally).

**Comment by ImportanceOfBeingErnest** [comment]
This was a PR (https://github.com/matplotlib/matplotlib/pull/14703). I can also reproduce this with current master, pyqt 5.9 on windows 8.

```
import matplotlib.pyplot as plt
fig = plt.figure(figsize=(5, 20))
print(fig.get_size_inches())
plt.show()
print(fig.get_size_inches())
```
outputs

```
[ 5. 20.]
[5.   6.66]
```

**Comment by colathro** [comment]
Also reproduced in Qt5 on MacOS.

```
import matplotlib
matplotlib.use('Qt5Agg')
import matplotlib.pyplot as plt

fig = plt.figure(figsize=(5, 20))

print(fig.get_size_inches())
plt.show()
print(fig.get_size_inches())
```

output
```
[ 5. 20.]
[5.   6.29]
```

**Comment by anntzer** [comment]
Sorry about that.  I wonder whether qt changed its sizing implementation recently?  I really thought I had tested that (as noted in #14703), but can repro the issue.  Will revert that PR.

**Comment by tacaswell** [comment]
Thanks for testing on master @larsoner !  The fix will make it in for 3.2.

**Comment by timhoffm** [comment]
Closed by #15613.
